modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainerV2.java

```diff
@@ -0,0 +1,176 @@
+package org.testcontainers.containers;
+
+import static org.junit.Assert.assertEquals;
+
+import com.influxdb.client.InfluxDBClient;
+import com.influxdb.client.InfluxDBClientFactory;
+import com.influxdb.client.InfluxDBClientOptions;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import lombok.SneakyThrows;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.containers.wait.strategy.WaitAllStrategy;
+import org.testcontainers.lifecycle.Startables;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.MountableFile;
+
+/**
+ * See
+ * <a href="https://docs.influxdata.com/influxdb/v2.0/get-started/#download-and-run-influxdb-v2-0">https://docs.influxdata.com/influxdb/v2.0/get-started/#download-and-run-influxdb-v2-0</a>
+ */
+public class InfluxDBContainerV2<SELF extends InfluxDBContainerV2<SELF>> extends GenericContainer<SELF> {
+
+    public static final Integer INFLUXDB_PORT = 8086;
+
+    private static final String REGISTRY = "quay.io";
+    private static final String REPOSITORY = "influxdb/influxdb";
+    private static final String TAG = "v2.0.0";
+    private static final DockerImageName DEFAULT_IMAGE_NAME =
+        DockerImageName.parse(String.format("%s/%s:%s", REGISTRY, REPOSITORY, TAG));
+    private static final int NO_CONTENT_STATUS_CODE = 204;
+    private static final String INFLUX_SETUP_SH = "influx-setup.sh";
+
+    private String username = "test-user";
+    private String password = "test-password";
+    private String bucket = "test-bucket";
+    private String organization = "test-org";
+    private int retention = 0;
+    private String retentionUnit = RetentionUnit.NANOSECONDS.label;
+
+    private InfluxDBContainerV2(final DockerImageName imageName) {
+        super(imageName);
+        imageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+        this.waitStrategy = (new WaitAllStrategy())
+            .withStrategy(Wait
+                .forHttp("/ping")
+                .withBasicCredentials(this.username, this.password)
+                .forStatusCode(NO_CONTENT_STATUS_CODE))
+            .withStrategy(Wait.forListeningPort());
+        this.addExposedPort(INFLUXDB_PORT);
+    }
+
+    public static InfluxDBContainerV2<?> createWithDefaultTag() {
+        return new InfluxDBContainerV2<>(DEFAULT_IMAGE_NAME);
+    }
+
+    public static InfluxDBContainerV2<?> createWithSpecificTag(final DockerImageName imageName) {
+        return new InfluxDBContainerV2<>(imageName);
+    }
```

Comment:
Please can we remove these? In all Testcontainers container classes, the constructor is our standard way of instantiation, and introducing a new style would be confusing and possibly constrain us when we evolve APIs.

Also we deliberately deprecated the idea of 'default tags' a while back because we found that they would become stale. The preferred style is a public constructor that accepts a `DockerImageName`, which puts the decision about which image to use into the user's hands.